contracts/hardhat.config.ts

```diff
@@ -0,0 +1,41 @@
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";
+
+const config: HardhatUserConfig = {
+  solidity: {
+    version: "0.8.18",
+    settings: {
+      // istanbul upgrade occurred before the london hardfork, so is compatible with kava's evm
+      evmVersion: "istanbul",
+      // optimize build for deployment to mainnet!
+      optimizer: {
+        enabled: true,
+        runs: 1000,
+      },
+    },
+  },
```

Comment:
here are the solc configurations and optimizations used to compile the contract.